Framework/PythonInterface/mantid/plots/datafunctions.py

```diff
@@ -423,27 +457,38 @@ def get_matrix_2d_ragged(workspace, normalize_by_bin_width, histogram2D=False, t
     delta = np.finfo(np.float64).max
     min_value = np.finfo(np.float64).max
     max_value = np.finfo(np.float64).min
+    try:
+        sp_info = workspace.spectrumInfo()
+    except:
+        sp_info = None
+
     for i in range(num_hist):
-        xtmp = workspace.readX(i)
-        if workspace.isHistogramData():
-            # input x is edges
-            xtmp = mantid.plots.datafunctions.points_from_boundaries(xtmp)
-        else:
-            # input x is centers
-            pass
-        min_value = min(min_value, xtmp.min())
-        max_value = max(max_value, xtmp.max())
-        diff = xtmp[1:] - xtmp[:-1]
-        delta = min(delta, diff.min())
+        if not(sp_info and sp_info.hasDetectors(i) and sp_info.isMonitor(i)):
+            xtmp = workspace.readX(i)
+            if workspace.isHistogramData():
+                # input x is edges
+                xtmp = mantid.plots.datafunctions.points_from_boundaries(xtmp)
+            else:
+                # input x is centers
+                pass
+            min_value = min(min_value, xtmp.min())
+            max_value = max(max_value, xtmp.max())
+            diff = xtmp[1:] - xtmp[:-1]
+            delta = min(delta, diff.min())
     num_edges = int(np.ceil((max_value - min_value)/delta)) + 1
     x_centers = np.linspace(min_value, max_value, num=num_edges)
     y = mantid.plots.datafunctions.boundaries_from_points(workspace.getAxis(1).extractValues())
     z = np.empty([num_hist, num_edges], dtype=np.float64)
+
     for i in range(num_hist):
         centers, ztmp, _, _ = mantid.plots.datafunctions.get_spectrum(
             workspace, i, normalize_by_bin_width=normalize_by_bin_width, withDy=False, withDx=False)
         f = interp1d(centers, ztmp, kind='nearest', bounds_error=False, fill_value=np.nan)
```

Comment:
what is f?

Reply:
if I understand correctly, f is the interpolating function used to make sure every histogram has the same width in the end